nemo/core/classes/exportable.py

```diff
@@ -153,34 +151,30 @@ def _export(
                         check_trace_input = [input_example]
                     else:
                         check_trace_input = check_trace
-
+                jitted_model = self
                 if format == ExportFormat.TORCHSCRIPT:
-
                     jitted_model = torch.jit.trace_module(
                         self,
                         {"forward": tuple(input_list) + tuple(input_dict.values())},
                         strict=True,
                         check_trace=check_trace,
                         check_tolerance=check_tolerance,
                     )
-                    if not self.training:
-                        jitted_model = torch.jit.optimize_for_inference(torch.jit.freeze(jitted_model))
+                    jitted_model = torch.jit.optimize_for_inference(torch.jit.freeze(jitted_model))
                     if verbose:
                         logging.info(f"JIT code:\n{jitted_model.code}")
                     jitted_model.save(output)
                     jitted_model = torch.jit.load(output)
 
                     if check_trace:
                         verify_torchscript(jitted_model, output, check_trace_input, input_names, check_tolerance)
-
                 elif format == ExportFormat.ONNX:
                     # dynamic axis is a mapping from input/output_name => list of "dynamic" indices
                     if dynamic_axes is None:
                         dynamic_axes = get_dynamic_axes(self.input_module.input_types, input_names)
                         dynamic_axes.update(get_dynamic_axes(self.output_module.output_types, output_names))
-
                     torch.onnx.export(
-                        self,
+                        jitted_model,
```

Comment:
use `self` here?

Reply:
This is to possibly use jit.trace() first. Now not working due to Torch bugs.